tools/cache_perf_tool/main.go

```diff
@@ -234,11 +294,14 @@ func mutexProfiling(ctx context.Context) {
 		case <-ticker.C:
 			fileName := fmt.Sprintf("mutex-%s.prof", time.Now().Format(time.RFC3339))
 			mutexProfBuf := bytes.Buffer{}
+			runtime.SetMutexProfileFraction(1)
+			time.Sleep(5 * time.Second)
```

Comment:
We can sleep for the interval here, to ensure the profile captures the full span - otherwise you can enable 1s profiling with a 5 second sleep which would not work correctly.

```suggestion
			time.Sleep(interval)
```